src/Pkg.jl

```diff
@@ -117,18 +117,34 @@ a package, also inside that package.
 The following table describes the argument values for `preserve` (in order of strictness):
 
-| Value             | Description                                                                         |
-|:------------------|:------------------------------------------------------------------------------------|
-| `PRESERVE_ALL`    | Preserve the state of all existing dependencies (including recursive dependencies)  |
-| `PRESERVE_DIRECT` | Preserve the state of all existing direct dependencies                              |
-| `PRESERVE_SEMVER` | Preserve semver-compatible versions of direct dependencies                          |
-| `PRESERVE_NONE`   | Do not attempt to preserve any version information                                  |
-| `PRESERVE_TIERED` | Use the tier which will preserve the most version information (this is the default) |
+| Value             | Description                                                                          |
+|:------------------|:-------------------------------------------------------------------------------------|
+| `PRESERVE_MAX`    | Preserve the state of all existing dependencies and only add those already installed |
+| `PRESERVE_ALL`    | Preserve the state of all existing dependencies (including recursive dependencies)   |
+| `PRESERVE_DIRECT` | Preserve the state of all existing direct dependencies                               |
+| `PRESERVE_SEMVER` | Preserve semver-compatible versions of direct dependencies                           |
+| `PRESERVE_NONE`   | Do not attempt to preserve any version information                                   |
+| `PRESERVE_TIERED` | Use the tier which will preserve the most version information (this is the default)  |
+
+!!! note
+    The `max` strategy will not be tried by the tiered approach unless the env var
+    `JULIA_PKG_PRESERVE_MAX` is enabled, or if the `allow_max` option is used.
+
+After the installation of new packages the project will be precompiled. For more information see `pkg> ?precompile`.
+
+With the `max` strategy the newly added packages will likely already be precompiled, but if not this may be
+because either the combination of package versions resolved in this environment has not been resolved and
+precompiled before, or the precompile cache has been deleted by the LRU cache storage
+(see JULIA_MAX_NUM_PRECOMPILE_FILES).
+
+!!! compat "Julia 1.9"
+    The `PRESERVE_MAX` strategy and `allow_max` option requires at least Julia 1.9.
 
 # Examples
 ```julia
 Pkg.add("Example") # Add a package from registry
-Pkg.add("Example"; preserve=Pkg.PRESERVE_ALL) # Add the `Example` package and preserve existing dependencies
+Pkg.add("Example"; preserve=Pkg.PRESERVE_ALL) # Add the `Example` package and strictly preserve existing dependencies
+Pkg.add("Example"; allow_max=true) # First try to add only the installed version of `Example` and its dependencies
```

Comment:
This sounds more permissive (more tolerant) than `PRESERVE_ALL`. I'm guessing it's actually less tolerant?

Reply:
Yeah the tricky thing is conveying it includes `all` AND the installed thing

Reply:
`allow_max` updated to simply `installed`,to match the `add --installed Example` form